doc/whats-new.rst

```diff
@@ -73,6 +77,8 @@ Internal Changes
 
 - Many error messages related to invalid dimensions or coordinates now always show the list of valid dims/coords (:pull:`8079`).
   By `András Gunyhó <https://github.com/mgunyho>`_.
+- Refactor of encoding and decoding times/timedeltas to preserve nanosecond resolution (:pull:`7827`).
```

Comment:
This was something we tried to accomplish in #4684 -- and I think we were successful for arrays that did not contain missing values (e.g. [this test](https://github.com/pydata/xarray/blob/8215911a8e1c97e58c3db071276a185b931b4a35/xarray/tests/test_coding_times.py#L1045-L1054)) -- but we overlooked this important issue related to arrays that did contain missing values.  We therefore might consider this a bug fix as well:
```suggestion
- Refactor of encoding and decoding times/timedeltas to preserve nanosecond resolution in arrays that contain missing values (:pull:`7827`).
```